### PR TITLE
feat: read only view for UI

### PR DESCRIPTION
**Pull request by veds-g**
- Added support to toggle Read Only View for Numaflow UI
- Update boolean `server.readonly` in `numaflow-cmd-params-config` to toggle view
- Added new endpoint - `/api/v1/readonlyinfo` to be used in the UI

image
image
image

### CODE REVIEW

**Comment by whynowy** [comment]
How to enable/disable it?

**Comment by whynowy** [comment]
> > How to enable/disable it?
> 
> today, default behaviour - current UI(edit), package(read). We need to modify a boolean variable in `App.tsx` component as mentioned in description above to enable/disable. Later, I am thinking of setting it up by accepting a variable from backend via API call, and set it up.

Maybe do it with an approach similar to the `localPath`?

**Comment by veds-g** [comment]
> Maybe do it with an approach similar to the `localPath`?

I tried doing it that way but I realised this approach won't work for the package. For `localPath` we create a script with config and directly inject it into the UI server. This works for standalone Numaflow but to solve both cases I'm thinking of writing a separate API. wdyt?

**Review by mshakira — APPROVED** [review]
looks good

File: cmd/commands/server.go [hunk 1]
```diff
@@ -69,6 +71,7 @@ func NewServerCommand() *cobra.Command {
 	command.Flags().BoolVar(&namespaced, "namespaced", sharedutil.LookupEnvBoolOr("NUMAFLOW_SERVER_NAMESPACED", false), "Whether to run in namespaced scope, defaults to false.")
 	command.Flags().StringVar(&managedNamespace, "managed-namespace", sharedutil.LookupEnvStringOr("NUMAFLOW_SERVER_MANAGED_NAMESPACE", sharedutil.LookupEnvStringOr("NAMESPACE", "numaflow-system")), "The namespace that the server watches when \"--namespaced\" is \"true\".")
 	command.Flags().StringVar(&baseHref, "base-href", sharedutil.LookupEnvStringOr("NUMAFLOW_SERVER_BASE_HREF", "/"), "Base href for Numaflow server, defaults to '/'.")
+	command.Flags().BoolVar(&readOnly, "readonly", sharedutil.LookupEnvBoolOr("NUMAFLOW_SERVER_READONLY", true), "Whether to enable read only view for the UX server, defaults to false.")
```
Comment (whynowy) [inline cmd/commands/server.go:74]: This means default is true.

File: server/cmd/server/start.go [hunk 2]
```diff
@@ -152,6 +156,7 @@ func UrlRewrite(r *gin.Engine) gin.HandlerFunc {
 func CreateAuthRouteMap(baseHref string) authz.RouteMap {
 	return authz.RouteMap{
 		"GET:" + baseHref + "api/v1/sysinfo":                                                         authz.NewRouteInfo(authz.ObjectPipeline, false),
+		"GET:" + baseHref + "api/v1/readonlyinfo":                                                    authz.NewRouteInfo(authz.ObjectEvents, false),
```
Comment (whynowy) [inline server/cmd/server/start.go:159]: Is it possible to put it into sysinfo and use that API?

File: cmd/commands/server.go (hunk 1 above)
Comment (veds-g) [reply cmd/commands/server.go:74]: typo, should be false

**Pull request merged**